pkg/utils/finalizer.go

```diff
@@ -0,0 +1,31 @@
+package utils
+
+import (
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const Finalizer = "kafkarator.kafka.nais.io"
+
+func AppendFinalizer(in client.Object) {
+	finalizers := in.GetFinalizers()
+	if finalizers == nil {
+		finalizers = make([]string, 0)
+	}
+	for _, v := range finalizers {
+		if v == Finalizer {
+			return
+		}
+	}
+	finalizers = append(finalizers, Finalizer)
+	in.SetFinalizers(finalizers)
+}
+
+func RemoveFinalizer(in client.Object) {
+	finalizers := make([]string, 0, len(in.GetFinalizers()))
+	for _, v := range in.GetFinalizers() {
+		if v != Finalizer {
+			finalizers = append(finalizers, v)
+		}
+	}
+	in.SetFinalizers(finalizers)
+}
```

Comment:
`controller-runtime` has util package for these: [`AddFinalizer`](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/controller/controllerutil#AddFinalizer) and [`RemoveFinalizer`](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/controller/controllerutil#RemoveFinalizer)

Reply:
Nice. Changed to use controllerutils now.